src/freenet/clients/fcp/RequestStatus.java

```diff
@@ -63,10 +69,15 @@ synchronized void restart(boolean started) {
 		this.totalBlocks = total;
 		this.minBlocks = min;
 		this.fetchedBlocks = fetched;
+        // clone() because Date is mutable
```

Comment:
I expect that this should reduce the maintenance burden in case people who don’t know much Java try to improve the codebase. It wouldn’t have been obvious to me that `clone()` is needed.